language-plutus-core/src/Language/PlutusCore/Evaluation/Machine/Cek.hs

```diff
@@ -55,48 +124,112 @@ type VarEnv = UniqueMap TermUnique Closure
 data CekEnv = CekEnv
     { _cekEnvMeans  :: DynamicBuiltinNameMeanings
     , _cekEnvVarEnv :: VarEnv
+    , _cekEnvProtocolParameters :: ()
     }
 
+data ExBudget = ExBudget { _exBudgetCPU :: ExCPU, _exBudgetMemory :: ExMemory }
+
+-- TODO memory should probably be fixed size - but then how do we handle overflow?
```

Comment:
`Int` vs. `Integer` - on one hand, `Int` guarantees space usage, but trades for danger of overflows. I'd prefer `Int64` with the evaluator blowing up on overflow.

Reply:
I'm happy with `Integer` - I don't think memory exhaustion due to the *budget* is a problem.

Reply:
Wouldn't dismiss it immediately, there's been weirder attacks.